src/OpenLoco/Ui/WindowManager.cpp

```diff
@@ -794,11 +794,44 @@ namespace OpenLoco::Ui::WindowManager
     // TODO: hook
     Window* bringToFront(Window* w)
     {
-        registers regs;
-        regs.esi = (uint32_t)w;
-        call(0x004CC750, regs);
+        if (w->flags & (WindowFlags::stick_to_back | WindowFlags::stick_to_front))
+        {
+            return w;
+        }
+
+        Window* frontMostWnd = nullptr;
+        for (auto i = count(); i != 0; --i)
+        {
+            if (!(_windows[i - 1].flags & WindowFlags::stick_to_back))
+            {
+                frontMostWnd = &_windows[i - 1];
+                break;
+            }
+        }
+        if (frontMostWnd != nullptr && frontMostWnd != &_windows[0] && frontMostWnd != w)
+        {
+            std::swap(*frontMostWnd, *w);
+            w = frontMostWnd;
+            w->invalidate();
+        }
+
+        auto right = w->x + w->width;
+        // If window is almost offscreen to the right
```

Comment:
This checks if the windows is almost off-screen on the left.

Reply:
Do you think we should add code for all the directions so that if it's of in any direction it will bring it in

Reply:
Honestly, I'm not sure. I can imagine people put windows slightly off-screen temporarily to just look at what's behind them. Also the news window slides in from the bottom, right? If we try to 'fix' that positioning as well, we'd break the animation.